contracts/margined_insurance_fund/src/state.rs

```diff
@@ -30,6 +30,63 @@ pub fn read_vammlist(deps: Deps, storage: &dyn Storage) -> StdResult<Vec<Addr>>
     keys
 }
 
+// function changes the bool stored under an address to 'false'
+// note that that means this can only be given an *existing* vamm
+pub fn vamm_off(deps: DepsMut, input: Addr) -> StdResult<()> {
+    // first check that there is data there
+    if VAMM_LIST.may_load(deps.storage, &input)?.is_none() {
+        return Err(StdError::GenericErr {
+            msg: "This vAMM has not been added".to_string(),
+        });
+    };
+    VAMM_LIST.save(deps.storage, &input, &false)
+}
+
+// function changes the bool stored under an address to 'true'
+// note that that means this can only be given an *existing* vamm
+pub fn vamm_on(deps: DepsMut, input: Addr) -> StdResult<()> {
+    // first check that there is data there
+    if VAMM_LIST.may_load(deps.storage, &input)?.is_none() {
+        return Err(StdError::GenericErr {
+            msg: "This vAMM has not been added".to_string(),
+        });
+    };
+    VAMM_LIST.save(deps.storage, &input, &true)
+}
+
+// this function reads the bool stored under an addr, and if there is no addr stored there then throws an error
+// use this function when you want to check the 'on/off' status of a vAMM
+pub fn read_vamm_status(storage: &dyn Storage, input: Addr) -> StdResult<bool> {
+    VAMM_LIST
+        .load(storage, &input)
+        .map_err(|_| StdError::GenericErr {
+            msg: "No vAMM stored".to_string(),
+        })
+}
+
+// this function reads the bools stored in the Map, and if there are no vAMMs stored, returns empty vec
+// use this function when you want to check the 'on/off' status of a vAMM
+pub fn read_all_vamm_status(storage: &dyn Storage) -> StdResult<Vec<(Addr, bool)>> {
+    let status_vec = VAMM_LIST
+        .range(storage, None, None, Order::Ascending)
+        .collect::<StdResult<Vec<(Vec<u8>, bool)>>>()?
+        .iter()
+        .map(|tup| {
+            (
+                Addr::unchecked(&String::from_utf8(tup.0.clone()).unwrap()),
+                tup.1,
+            )
+        })
+        .collect();
```

Comment:
This code block could also be improved (I think) - again the .range() in cw-storage-plus 0.8.0 doesn't return nice values (like the .keys() method)

Reply:
Yeah I think that'll be my next step - the option to shutdown a custom range of vAMMs and to query a custom range of vAMMs